test/extended/openstack/volumes.go

```diff
@@ -180,6 +184,67 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack pla
 				return nil
 			}, "60s", "10s").Should(o.BeNil())
 		})
+
+		g.It("should create a manila share when using manila storage class", func() {
+			var err error
+			manilaSc := FindStorageClassByProvider(oc, "manila.csi.openstack.org")
+			if manilaSc == nil {
+				e2eskipper.Skipf("No StorageClass with manila.csi.openstack.org provisioner")
+			}
+
+			ns := oc.Namespace()
+			pvc := CreatePVC(ctx, clientSet, "manila-pvc", ns, manilaSc.Name, "1Gi")
+			fileContent := "hello"
+
+			// Skip if Manila Storage class is not defined
+
+			shareClient, err = client("sharev2")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			// Make sure a Manila share was created with the same name as the PVC
+			err = waitPvcVolume(ctx, clientSet, pvc.Name, ns)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			manilaShares, err := GetSharesFromName(shareClient, pvc.Spec.VolumeName)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			//			_, err := shares.Get(shareClient, pvc.Spec.VolumeName).Extract()
+			//			manilaShares, err := GetSharesFromName(shareClient, pvc.Spec.VolumeName)
+			//			o.Expect(err).NotTo(o.HaveOccurred())
+			//			e2e.Logf("PVC: %v, Shares: %v", pvc.Spec.VolumeName, manilaShares)
+			shareID := manilaShares[0].ID
+			_, err = shares.Get(shareClient, shareID).Extract()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Creating a deployment with the volumes attached
+			g.By("Creating Openshift deployment")
+			labels := map[string]string{"app": "manila-test-dep"}
+
+			dep := NewDeploymentOpts()
+			dep.Name = "manila-test-dep"
+			dep.Labels = labels
+			volOpt := volumeOption{
+				Name:      "data-volume",
+				PvcName:   pvc.Name,
+				MountPath: "data",
+			}
+			dep.Volumes = append(dep.Volumes, volOpt)
+
+			//testDeployment := createTestDeployment("manila-test-dep", labels, 2, "TCP", 8080, "data-volume", pvc.Name, "data")
+			testDeployment := createTestDeployment(dep)
+
+			deployment, err := clientSet.AppsV1().Deployments(ns).Create(ctx,
+				testDeployment, metav1.CreateOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = e2edeployment.WaitForDeploymentComplete(clientSet, deployment)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			pods, err := oc.AdminKubeClient().CoreV1().Pods(ns).List(ctx, metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Create a file in one pod's attached volume and read it in the second one
+			_, err = oc.Run("exec").Args(pods.Items[0].Name, "--", "bash", "-c", fmt.Sprintf("echo %s > /data/1", fileContent)).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			out, err := oc.Run("exec").Args(pods.Items[0].Name, "--", "bash", "-c", "cat /data/1").Output()
```

Comment:
I think you need to check that what you write in pod 0 is accesible in pod 1, as both have the same PVC mounted, right? That's the utility of manila(RWX).
```suggestion
			out, err := oc.Run("exec").Args(pods.Items[1].Name, "--", "bash", "-c", "cat /data/1").Output()
```

Reply:
Right. That was my intention :facepalm: